### DISCUSSION TITLE
Limiting AWS access rights for DescribeInstances to specific instances instead of *

### DISCUSSION THREAD

**Discussion by Wlkus** [BohdanPetryshyn/basti#119] [Q&A] [answered]
We have different EC2 instances for multiple projects in the AWS account and allowing Basti to have access to * with ec2:DescribeInstances doesn't seem to be right from security perspective - but maybe I am wrong? Is there any guide to allow it selectively only to instances which Basti really uses?

As described [here](https://github.com/basti-app/basti?tab=readme-ov-file#minimal-iam-permissions)
    {
      "Effect": "Allow",
      "Action": "ec2:DescribeInstances",
      "Resource": "*"
    },

If I limit resource instead of the asterisk only to exact t3.micro instance running basti in aws, then it ends up with "Error checking target state. Access denied by IAM"
Using * of course works, but I assume then this allows access for DescribeInstances function to all the EC2 machines on the same account?

**Comment by BohdanPetryshyn** [comment, marked as answer]
Hi @Wlkus, thank you for your question! When developing Basti, I always try to make IAM statements as restrictive as possible. Unfortunately, in the case of `ec2:DescribeInstances`, resource-level restrictions are not supported by AWS ([docs](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonec2.html#amazonec2-DescribeInstanceStatus)).
